BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -106,17 +114,10 @@ - (NSURLSession *)createBackgroundSession
 
 #pragma mark - public methods
 
-- (void)setUpWithDefaultDelegate:(id<BOXURLSessionManagerDelegate>)delegate
+- (void)setUpToSupportBackgroundTasksWithDefaultDelegate:(id<BOXURLSessionManagerDelegate>)delegate
 {
-    @synchronized (self) {
-        if (self.hasCompletedSetup == NO) {
-            self.defaultDelegate = delegate;
-            self.defaultSession = [self createDefaultSession];
-            self.progressSession = [self createProgressSession];
-            self.backgroundSession = [self createBackgroundSession];
-            self.hasCompletedSetup = YES;
-        }
-    }
+    self.defaultDelegate = delegate;
+    [self createBackgroundSession];
```

Comment:
it's safe since we check for _backgroundSession == nil first, and NSURLSession will always give us the same instance of that background session of a given id after it was created